include/seastar/core/io_queue.hh

```diff
@@ -172,7 +171,7 @@ public:
     void request_dispatched() noexcept;
     void cancel_request(queued_io_request& req) noexcept;
     void complete_cancelled_request(queued_io_request& req) noexcept;
-    void complete_request(io_desc_read_write& desc) noexcept;
+    void complete_request(queued_io_request& req) noexcept;
```

Comment:
That's because upon completion fair-queue needs to be notified about completed request. And this, in turn, is now empty but used to be needed because fair-queue erroneously contained the output throttler on board. In fact, the throttler should be on its own and referenced by io_queue directly, it's not the part of fair queue

Reply:
Is `fair_queue::notify_request_finished()` needed then? As you said it does nothing. Do you think that the empty function can be safely removed?

If it was removed, then we could rename `io_queue::complete_request(queued_io_request& req)` to `io_queue::request_completed()` -- the function would only be responsible for accounting -- just like `io_queue::request_dispatched()`.


Moreover, I was wondering if we want to rename (or somehow rework) `io_queue::queue_request()` and `io_queue::queue_one_request()` . Both functions are specific to read and write requests -- they explicitly create `queued_read_write_request`.

I just think out loud -- perhaps, the discards request would not need to take `internal::io_request` parameter -- it is needed by `io_sink`.